src/acl_kernel_if.cpp

```diff
@@ -817,10 +817,8 @@ int acl_kernel_if_update(const acl_device_def_autodiscovery_t &devdef,
           devdef.accel[ii].iface.name.c_str(), kern->accel_csr[ii].address,
           kern->accel_csr[ii].bytes);
     }
-    if (acl_platform.offline_mode == ACL_CONTEXT_MPSIM) {
```

Comment:
What the benefit of moving this if statement inside of the `acl_hal_mmd_simulation_set_kernel_cra_address_map` function? It's kind of strange that always calling this function but it just does nothing when the mode is not simulation though.

Reply:
Good question! As per Joe's suggestion the runtime should be agnostic to whether we are running in simulation or not, so this is a preparation step for the removal of the ACL_CONTEXT_MPSIM mode altogether.